Build Error on OS X 10.9.3

I tried to install gfx.js with the command
"
sudo luarocks install https://raw.github.com/clementfarabet/gfx.js/master/gfx.js-scm-0.rockspec
"
but it returns with errors:
==> installing Node.js dependencies

> pty.js@0.2.4 install /private/tmp/luarocks_gfx.js-scm-0-1414/gfx.js/node_modules/pty.js
> node-gyp rebuild

  CXX(target) Release/obj.target/pty/src/unix/pty.o
../src/unix/pty.cc:487:10: error: use of undeclared identifier 'openpty'
  return openpty(amaster, aslave, name, (termios _)termp, (winsize *)winp);
         ^
../src/unix/pty.cc:533:10: error: use of undeclared identifier 'forkpty'
  return forkpty(amaster, name, (termios *)termp, (winsize *)winp);
         ^
2 errors generated.
make: *_\* [Release/obj.target/pty/src/unix/pty.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:809:12)
gyp ERR! System Darwin 13.2.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /private/tmp/luarocks_gfx.js-scm-0-1414/gfx.js/node_modules/pty.js
gyp ERR! node -v v0.10.29
gyp ERR! node-gyp -v v0.13.1
gyp ERR! not ok 

> ws@0.4.31 install /private/tmp/luarocks_gfx.js-scm-0-1414/gfx.js/node_modules/socket.io/node_modules/socket.io-client/node_modules/ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)

  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/bufferutil.node
  SOLINK_MODULE(target) Release/bufferutil.node: Finished
  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/validation.node
  SOLINK_MODULE(target) Release/validation.node: Finished
npm ERR! pty.js@0.2.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pty.js@0.2.4 install script.
npm ERR! This is most likely a problem with the pty.js package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls pty.js
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 13.2.0
npm ERR! command "node" "/usr/local/bin/npm" "install"
npm ERR! cwd /private/tmp/luarocks_gfx.js-scm-0-1414/gfx.js
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.14
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /private/tmp/luarocks_gfx.js-scm-0-1414/gfx.js/npm-debug.log
npm ERR! not ok code 0
==> installing static resources into ~/.gfx.js/
==> installing torch client into /usr/local/bin/../share/lua/5.1/gfx/

It was able to work but today when I tried to reinstalled it, it failed...

Thanks!

I have command line tools and still have problems with pty.js

npm ERR! pty.js@0.2.4 install: `node-gyp rebuild`

npm ERR! Exit status 1

npm ERR!

npm ERR! Failed at the pty.js@0.2.4 install script.

npm ERR! This is most likely a problem with the pty.js package,

npm ERR! not with npm itself.

npm ERR! Tell the author that this fails on your system:

npm ERR!     node-gyp rebuild

npm ERR! You can get their info via:

npm ERR!     npm owner ls pty.js

npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.2.0

npm ERR! command "node" "/usr/local/bin/npm" "install"

npm ERR! cwd /private/tmp/luarocks_gfx.js-scm-0-3979/gfx.js

npm ERR! node -v v0.10.29

npm ERR! npm -v 1.4.14

npm ERR! code ELIFECYCLE

npm ERR!

npm ERR! Additional logging details can be found in:

npm ERR!     /private/tmp/luarocks_gfx.js-scm-0-3979/gfx.js/npm-debug.log

npm ERR! not ok code 0

Thank you for your help.

-Andrew

On Sat, Jun 28, 2014 at 1:09 PM, Soumith Chintala notifications@github.com
wrote:

> Hey, apparently OSX doesn't ship with the relevant headers, and none of us
> noticed because we have XCode Commandline Tools installed.
> atom/terminal#28 (comment)
> https://github.com/atom/terminal/issues/28#issuecomment-39405908
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/clementfarabet/gfx.js/issues/17#issuecomment-47433017
> .

## 

Andrew Hartnett
Couzin Lab
508-566-2206
ahartnet@princeton.edu

When trying to run luajit -lgfx.go
I get the error below.

module.js:340
    throw err;
          ^
Error: Cannot find module '../build/Release/pty.node'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/ronanmoynihan/.gfx.js/node_modules/pty.js/lib/pty.js:10:11)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "dependencies": {
         "express": "3.4.4",
         "socket.io": "1.1.0",
-        "pty.js": "0.2.4",
+        "pty.js": "soumith/pty.js",
         "recursive-readdir": "0.0.2"
     },
     "optionalDependencies" : {

